---
layout: tool 
title: Edgar Synteny Matrix
developer_github_account: 
paper: https://www.ncbi.nlm.nih.gov/pubmed/15215394
paper_title: "ISTA: computational tools for comparative genomics."
doi: 10.1093/nar/gkh458
browser: http://genome.lbl.gov/vista/index.shtml
abstract: "Comparison of DNA sequences from different species is a fundamental method for identifying functional elements in genomes. Here, we describe the VISTA family of tools created to assist biologists in carrying out this task. Our first VISTA server at http://www-gsd.lbl.gov/vista/ was launched in the summer of 2000 and was designed to align long genomic sequences and visualize these alignments with associated functional annotations. Currently the VISTA site includes multiple comparative genomics tools and provides users with rich capabilities to browse pre-computed whole-genome alignments of large vertebrate genomes and other groups of organisms with VISTA Browser, to submit their own sequences of interest to several VISTA servers for various types of comparative analysis and to obtain detailed comparative analysis results for a set of cardiovascular genes. We illustrate capabilities of the VISTA site by the analysis of a 180 kb interval on human chromosome 5 that encodes for the kinesin family member 3A (KIF3A) protein."
citation: "Frazer KA, Pachter L, Poliakov A, Rubin EM, Dubchak I. VISTA: computational tools for comparative genomics. Nucleic Acids Res. academic.oup.com; 2004;32: W273â€“9."
citation_count: 353
type: ['point sparse']
interconnection_type: ['between']
genome_layout: ['linear']
partition: ['contiguous']
abstraction: ['no']
arrangement: linear orthogona;
view: multiple
scale: single
focus: single
coordinate_systems: multiple
access_format: [web application]
user_documentation_availability: yes
user_documentation: https://www.uni-giessen.de/fbz/fb08/Inst/bioinformatik/software/EDGAR/documentation
pub_year: 2016
license: unavailable
license_form: unknown
supported_files: ['other']
image: /assets/edgarsyntenymatrix.png
image_location: https://edgar.computational.bio.uni-giessen.de/cgi-bin/edgar.cgi
notes: 
---
